### PR TITLE
 Requires.NotNull-->ArgumentNullException.ThrowIfNull

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary
Replace all `NotNull` with `ArgumentNullException.ThrowIfNull` excluding ComInterop code

## PR Context
As discussed in #18792

The new API [ArgumentNullException.ThrowIfNull](https://learn.microsoft.com/en-us/dotnet/api/system.argumentnullexception.throwifnull?view=net-7.0#system-argumentnullexception-throwifnull(system-object-system-string)) allows you to omit the parameter name, in which case it will use the name of the argument passed in. The passed in argument usually is the parameter to be validated, so that's just the right name to use when calling `ArgumentNullException.ThrowIfNull(argument)`.

### CODE REVIEW

File: src/System.Management.Automation/engine/Subsystem/PredictionSubsystem/CommandPrediction.cs [hunk 1]
```diff
@@ -142,7 +142,7 @@ public static class CommandPrediction
         /// <param name="history">History command lines provided as references for prediction.</param>
         public static void OnCommandLineAccepted(PredictionClient client, IReadOnlyList<string> history)
         {
-            Requires.NotNull(history, nameof(history));
+            ArgumentNullException.ThrowIfNull(history, nameof(history));
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/Subsystem/PredictionSubsystem/CommandPrediction.cs:145]: No need to have second argument. Below too.

**Review by iSazonov — APPROVED** [review]

**Review by xtqqczze — APPROVED** [review]
LGTM

File: src/System.Management.Automation/engine/ComInterop/IDispatchMetaObject.cs [hunk 2]
```diff
@@ -23,7 +23,7 @@ internal IDispatchMetaObject(Expression expression, IDispatchComObject self)
 
         public override DynamicMetaObject BindInvokeMember(InvokeMemberBinder binder, DynamicMetaObject[] args)
         {
-            Requires.NotNull(binder, nameof(binder));
+            ArgumentNullException.ThrowIfNull(binder);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/ComInterop/IDispatchMetaObject.cs:26]: The file is copied from .Net Runtime repo. Please keep it unchanged.

Comment (CarloToso) [reply src/System.Management.Automation/engine/ComInterop/IDispatchMetaObject.cs]: reverted

**Review by iSazonov — APPROVED** [review]

File: src/System.Management.Automation/engine/ComInterop/BoundDispEvent.cs [hunk 3]
```diff
@@ -73,7 +73,7 @@ private static void VerifyHandler(object handler)
         /// <returns>The original event with handler added.</returns>
         private object InPlaceAdd(object handler)
         {
-            Requires.NotNull(handler, nameof(handler));
+            ArgumentNullException.ThrowIfNull(handler);
```
Comment (iSazonov) [inline src/System.Management.Automation/engine/ComInterop/BoundDispEvent.cs:76]: The file please too.

**Review by iSazonov — APPROVED** [review]

**Pull request merged**